workflow/artifacts/http/http.go

```diff
@@ -72,9 +72,41 @@ func (h *ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 	return err
 }
 
-func (h *ArtifactDriver) OpenStream(a *wfv1.Artifact) (io.ReadCloser, error) {
-	// todo: this is a temporary implementation which loads file to disk first
-	return common.LoadToStream(a, h)
+func (h *ArtifactDriver) OpenStream(inputArtifact *wfv1.Artifact) (io.ReadCloser, error) {
+	var req *http.Request
+	var url string
+	var err error
+	if inputArtifact.Artifactory != nil && inputArtifact.HTTP == nil {
+		url = inputArtifact.Artifactory.URL
+		req, err = http.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			return nil, err
+		}
+		req.SetBasicAuth(h.Username, h.Password)
+	} else {
+		url = inputArtifact.HTTP.URL
```

Comment:
just looking at this code it seems like HTTP can be nil and Artifactory can be nil on this line - should we add a case for that even if currently maybe it's not possible to enter this code for that case? (in case it is in the future)

Reply:
Thanks. Added